homeassistant/components/sql/sensor.py

```diff
@@ -87,7 +87,7 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
             value_template.hass = hass
 
         # MSSQL uses TOP and not LIMIT
-        if not ("LIMIT" in query_str or "SELECT TOP" in query_str):
+        if not ("LIMIT" in query_str.upper() or "SELECT TOP" in query_str.upper()):
```

Comment:
I would definitely recommend adding a test that reproduces the problem, as i'm not sure if this change would pass that test.

This seems like it works for this if statement check, but reading the next line I think the replace may not work -- if i understand right?

Doing an .upper() on the query to replace it may change the final query so not sure that works below.

Reply:
lower case is allowed in the final query, just need to check if limit already exists

test added

Reply:
I guess the problem case is just that `select 5 as value` doesn't get changed to `SELECT TOP 5 as value` -- is that fine?

Reply:
that is a different issue indeed from https://github.com/home-assistant/core/pull/42778

Since then we have also dropped support for MSSQL https://github.com/home-assistant/architecture/blob/master/adr/0018-supported-databases.md

But I don't feel like dropping it right now...